### ISSUE TITLE
Corrupted labels after using different cameras?

### ISSUE THREAD

**Issue by st162053** [ultralytics/yolov5#6383]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi everyone, 
I use Yolov5 to detect custom objects and train on Google colab.
**My issue:** 
I used two different cameras (**Iphone and Canon**)  for collecting data and labelled my data using makesense.ai. All my labels for the Iphone pictures (size: 3024x4032 ) seem to be corrupted when trying to train, while my camera pictures (6000x4000) work just fine and custom training works. Is there a way to fix it and use my entire annotated datsaet for custom training? 
Thank you in advance!

![Screenshot (1)](https://user-images.githubusercontent.com/94604405/150540419-cadcf929-b4a5-47fc-b188-0ddb660326e1.png)

**Comment by glenn-jocher** [comment]
@st162053 what's your workflow like for your iPhone images?

From iOS to macOS using airdrop I'll get HEIC format which is Apple's proprietary H.265 implementation. On export to JPG everything trains normally for me.

**Comment by glenn-jocher** [comment]
@st162053 also you may want to update your PIL package to see if that fixes it.

**Comment by st162053** [comment]
Thanks! 
I took pictures with my IPhone 11 and transferred them to my Windows computer using USB. They were exported as JPG and then labelled and annotated manually. 
Trying PIL upgrade now!

**Comment by st162053** [comment]
The labels are no longer marked as corrupted but as missing, when using my pictures from both cameras. It works completely fine with only camera pictures.
![Screenshot (3)](https://user-images.githubusercontent.com/94604405/150643878-b0a8984b-7f8b-4663-99d0-88fd00506817.png)
![Screenshot (2)](https://user-images.githubusercontent.com/94604405/150643928-753ffe08-22e0-44b8-b33b-d318b6b4f52f.png)

**Comment by glenn-jocher** [comment]
@st162053 if jpegs are corrupt they will be restored, all you have to do is run the training command again.

If there are no labels then you've structured your dataset incorrectly and YOLOv5 can't find the labels *.txt files. To train correctly your data must be in YOLOv5 format. Please see our [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) tutorial for full documentation on dataset setup and all steps required to start training your first model. A few excerpts from the tutorial:

### 1.1 Create dataset.yaml

[COCO128](https://www.kaggle.com/datasets/ultralytics/coco128) is an example small tutorial dataset composed of the first 128 images in [COCO](http://cocodataset.org/#home) train2017. These same 128 images are used for both training and validation to verify our training pipeline is capable of overfitting. [data/coco128.yaml](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), shown below, is the dataset config file that defines 1) the dataset root directory `path` and relative paths to `train` / `val` / `test` image directories (or *.txt files with image paths), 2) the number of classes `nc` and 3) a list of class `names`:
```yaml
# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ../datasets/coco128  # dataset root dir
train: images/train2017  # train images (relative to 'path') 128 images
val: images/train2017  # val images (relative to 'path') 128 images
test:  # test images (optional)

# Classes
nc: 80  # number of classes
names: [ 'person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light',
         'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow',
         'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee',
         'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
         'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
         'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch',
         'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone',
         'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear',
         'hair drier', 'toothbrush' ]  # class names
```


### 1.2 Create Labels

After using a tool like [Roboflow Annotate](https://roboflow.com/annotate?ref=ultralytics) to label your images, export your labels to **YOLO format**, with one `*.txt` file per image (if no objects in image, no `*.txt` file is required). The `*.txt` file specifications are:

- One row per object
- Each row is `class x_center y_center width height` format.
- Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and `width` by image width, and `y_center` and `height` by image height.
- Class numbers are zero-indexed (start from 0).

<img width="800" alt="Image Labels" src="https://user-images.githubusercontent.com/26833433/91506361-c7965000-e886-11ea-8291-c72b98c25eec.jpg">

The label file corresponding to the above image contains 2 persons (class `0`) and a tie (class `27`):

<p align="center"><img width="428" src="https://user-images.githubusercontent.com/26833433/112467037-d2568c00-8d66-11eb-8796-55402ac0d62f.png"></p>


### 1.3 Organize Directories

Organize your train and val images and labels according to the example below. YOLOv5 assumes  `/coco128` is inside a `/datasets` directory **next to** the `/yolov5` directory. **YOLOv5 locates labels automatically for each image** by replacing the last instance of `/images/` in each image path with `/labels/`. For example:
```bash
../datasets/coco128/images/im0.jpg  # image
../datasets/coco128/labels/im0.txt  # label
```

<p align="center"><img width="700" src="https://user-images.githubusercontent.com/26833433/134436012-65111ad1-9541-4853-81a6-f19a3468b75f.png"></p>
</details>


Good luck 🍀 and let us know if you have any other questions!
